StudyHub/StudyHub.DAL/Repositories/Repository.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Microsoft.EntityFrameworkCore;
+using StudyHub.DAL.EF;
+using StudyHub.DAL.Repositories.Interfaces;
+using System.Collections;
+using System.Linq.Expressions;
+
+namespace StudyHub.DAL.Repositories;
+public class Repository<TEntity> : IRepository<TEntity>
+    where TEntity : class
```

Comment:
maybe better to have `TEntity : EntityBase` to prevent using any class with repository?